src/parse.jl

```diff
@@ -0,0 +1,227 @@
+    end
+    return x, code
+end
+function _divpow10!(x::BigInt, code, pow, ::RoundingMode{:ToZero})
+    @inbounds y = Parsers.access_threaded(() -> (@static VERSION > v"1.5" ? BigInt(; nbits=256) : BigInt()), _BIGINT_10s) # we must not yield here!
+    Base.GMP.MPZ.set!(y, _BIGINT10) # y = 10
+    Base.GMP.MPZ.pow_ui!(y, pow)    # y = y^pow
+    Base.GMP.MPZ.tdiv_q!(x, y)      # x = div(x, y)
+    return x, code
+end
+
+function _divpow10!(x::BigInt, code, pow, ::RoundingMode{:Throw})
+    @inbounds y = Parsers.access_threaded(() -> (@static VERSION > v"1.5" ? BigInt(; nbits=256) : BigInt()), _BIGINT_10s) # we must not yield here!
+    Base.GMP.MPZ.set!(y, _BIGINT10)   # y = 10
+    Base.GMP.MPZ.pow_ui!(y, pow)      # y = y^pow
+    Base.GMP.MPZ.tdiv_qr!(x, y, x, y) # x, y = divrem(x, y)
+    y == 0 || (code |= Parsers.INEXACT)
+    return x, code
+end
+
+# Rescale the digits we accumulated so far into the the a an integer representing the decimal
+@inline function Parsers.scale(
+    conf::FixedDecimalConf{T}, ::Parsers.FloatType, digits::V, exp, neg, code, ndigits, f::F, options::Parsers.Options
+) where {T,V,F}
+    rounding = something(options.rounding, RoundThrows)
+    # Positive: how many trailing zeroes we need to add to out integer
+    # Negative: how many digits are past our precision (we need to handle them in rounding)
+    decimal_shift = conf.f + exp
+    # Number of digits we need to accumulate including any trailigng zeros or digits past our precision
+    backing_integer_digits = ndigits + decimal_shift
+    may_overflow = backing_integer_digits == _maxintdigits(T)
+    if iszero(ndigits)
+        # all digits are zero
+        i = zero(T)
+    elseif backing_integer_digits < 0
+        # All digits are past our precision, no overflow possible
+        i = zero(T)
+        (rounding === RoundThrows) && (code |= Parsers.INEXACT)
+    elseif neg && (T <: Unsigned)
+        # Unsigned types can't represent negative numbers
+        i = _unsafe_convert_int(T, digits)
+        code |= Parsers.INVALID
+    elseif backing_integer_digits > _maxintdigits(T)
+        i = _unsafe_convert_int(T, digits)
+        # The number of digits to accumulate is larger than the capacity of T, we overflow
+        # We don't check for inexact here because we already have an error
+        code |= Parsers.OVERFLOW
+    else
+        if decimal_shift > 0
+            r = _unsafe_convert_int(T, digits)
+            i = _shift(r, decimal_shift)
+            may_overflow && (r >= i) && (code |= Parsers.OVERFLOW)
+        elseif decimal_shift < 0
+            if rounding === RoundNearest
+                r, code = _divpow10!(digits, code, -decimal_shift, RoundNearest)
+            elseif rounding === RoundToZero
+                r, code = _divpow10!(digits, code, -decimal_shift, RoundToZero)
+            else
+                r, code = _divpow10!(digits, code, -decimal_shift, RoundThrows)
+            end
```

Comment:
Shouldn't it be just as type-stable/performant to just do:
```suggestion
            r, code = _divpow10!(digits, code, -decimal_shift, rounding)
```
here?

Reply:
No, that would cause dynamic dispatch